### PR TITLE
add guide to contributing

### PR DESCRIPTION
**Pull request by des-des**
As discussed in #54. This is a first draft for a contributing guide.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Inspired by [dwys's contribution workflow](https://www.github.com/dwyl/contributing).
+
+# Contribuation guidelines
```
Comment (bradreeder) [inline CONTRIBUTING.md]: Typo, should be 'Contribution'

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Before starting work
+
+ 1. Search this repo's [issues](https://www.github.com/foundersandcoders/master-reference/issues) to see if an issue exists for the problem you are solving
+ 2. If the issue does not exist, create it.
+ 3. Indicate on the issue that you would like to take it on. Check the [readme](https://github.com/foundersandcoders/master-reference/blob/master/README.md) find someone to `@` if you need a response.
```
Comment (bradreeder) [inline CONTRIBUTING.md]: Typo, should be 'Check the readme and find someone'

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+related #[issue number]
+```
+or
+```md
+short description under 50 chars.
+[newline]
+related #[issue number]
+```
+
+It is important the you reference an issue in each commit.
```
Comment (bradreeder) [inline CONTRIBUTING.md]: Typo, should be 'It is important that you'.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+## Pull Requests
+
+Once you have finished your work, push up your branch and make a pull request. Make sure it has the following
+
+  1. A good title
+  2. Description, with detail of everything in the pull.
+  3. Reference to the issue in description.
+  3. An assignee, assign someone from [here](https://github.com/foundersandcoders/master-reference/blob/master/README.md
+
+# Thankyou!
```
Comment (bradreeder) [inline CONTRIBUTING.md]: I think 'thankyou' is a noun like 'a thankyou card' and that as a verb it's 'thank you'.

File: CONTRIBUTING.md (hunk 1 above)
Comment (bradreeder) [inline CONTRIBUTING.md]: + first line should be 'dwyl's'

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+Inspired by [dwys's contribution workflow](https://www.github.com/dwyl/contributing).
```
Comment (nelsonic) [inline CONTRIBUTING.md]: ❤️  U @des-des!
Hope you're having a superb 2017 (_so far_)!
❤️  ✅  🚀

Comment (iteles) [inline CONTRIBUTING.md]: <3

**Pull request merged**